src/containers/OrderReviewPage.jsx

```diff
@@ -174,31 +177,125 @@ class OrderReviewPage extends React.Component {
+      };
+
+      const zoomArrayProfile = [
+        'defaultcart',
+        'defaultcart:total',
+        'defaultprofile',
+        'defaultprofile:emails',
+        'defaultprofile:emails:element',
+      ];
+      const addressLink = orderData._order[0]._deliveries[0]._element[0]._destinationinfo[0]._destination[0].self.uri;
+      login()
+        .then(() => {
+          cortexFetch(addressLink, {
+            headers: {
+              'Content-Type': 'application/json',
+              Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+            },
+          })
+            .then(billingData => billingData.json())
+            .then((billingData) => {
+              bodyCybersourceRequest = {
+                ...bodyCybersourceRequest,
+                bill_to_address_line1: billingData.address['street-address'],
+                bill_to_address_city: billingData.address.locality,
+                bill_to_address_state: billingData.address.region,
+                bill_to_address_country: billingData.address['country-name'],
+                bill_to_address_postal_code: billingData.address['postal-code'],
+              };
+              cortexFetch(`/?zoom=${zoomArrayProfile.join()}`,
+                {
+                  headers: {
+                    'Content-Type': 'application/json',
+                    Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+                  },
+                })
+                .then(profileData => profileData.json())
+                .then((profileData) => {
+                  bodyCybersourceRequest = {
+                    ...bodyCybersourceRequest,
+                    currency: profileData._defaultcart[0]._total[0].cost[0].currency,
+                    bill_to_email: profileData._defaultprofile[0]._emails[0]._element[0].email,
+                  };
+                  fetch(`${uriForRequest}`, {
+                    method: 'POST',
+                    headers: {
+                      'Content-Type': 'application/json',
+                    },
+                    body: JSON.stringify(bodyCybersourceRequest),
+                  })
+                    .then(cybersourceData => cybersourceData.json())
+                    .then((cybersourceData) => {
+                      // eslint-disable-next-line
+                      const formEncodedBody = Object.keys(cybersourceData).filter(k => cybersourceData.hasOwnProperty(k)).map(k => encodeURIComponent(k) + '=' + encodeURIComponent(cybersourceData[k])).join('&');
+                      fetch(`${uriForCreateToken}`, {
+                        method: 'POST',
+                        headers: {
+                          'Content-Type': 'application/x-www-form-urlencoded',
+                        },
+                        body: formEncodedBody,
+                      })
+                        .then(data => data.text())
+                        .then((data) => {
```

Comment:
This should be much cleaner with the usage of the lambda. What's the reason for parsing html?